stablehlo/reference/Ops.cpp

```diff
@@ -178,6 +178,22 @@ Tensor eval(ReshapeOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(ReverseOp op, const Tensor &operand) {
+  Tensor result(op.getType());
+  auto shape = op.getType().getShape();
+  SmallVector<int64_t> dimensions;
+  for (auto dim : op.getDimensions()) dimensions.push_back(dim.getSExtValue());
+  for (auto resultIt = result.index_begin(), operandIt = operand.index_begin();
+       resultIt != result.index_end(); ++resultIt, ++operandIt) {
+    auto operandIdx = (*operandIt).vec();
+    for (auto dim : dimensions) {
+      operandIdx[dim] = -(operandIdx[dim] - (shape[dim] - 1));
+    }
+    result.set(*resultIt, operand.get(operandIdx));
+  }
+  return result;
+}
+
```

Comment:
Suggestion:
1. Avoid loop in collecting the dimensions in a `SmallVector`.
2. As per the spec, given an index of result we can derive the index of the operand. With that said, 
  -  we can avoid the second iterator. 
  -  make the implementation follow more closely to the spec
3. Prefer `SmallVector` over `std::vector`

```
Tensor eval(ReverseOp op, const Tensor &operand) {
  Tensor result(op.getType());
  auto shape = op.getType().getShape();
  auto dimensions = op.getDimensions().getValues<int64_t>();

  for (auto resultIt = result.index_begin(); resultIt != result.index_end();
       ++resultIt) {
    SmallVector<int64_t> operandIdx(*resultIt);
    for (auto dim : dimensions) {
      operandIdx[dim] = (shape[dim] - 1) - operandIdx[dim];
    }
    result.set(*resultIt, operand.get(operandIdx));
  }
  return result;
}

```